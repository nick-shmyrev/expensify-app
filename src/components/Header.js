import { NavLink } from 'react-router-dom';
import React from 'react';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <nav>
      <NavLink to="/" exact activeClassName="active">Dashboard</NavLink>
      <NavLink to="/create" activeClassName="active">Create Expense</NavLink>
      <NavLink to="/help" activeClassName="active">Help Page</NavLink>
    </nav>
  </header>
);

export default Header;
