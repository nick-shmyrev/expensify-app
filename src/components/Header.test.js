import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from './Header';

describe('Header component', () => {
  test('should render correctly', () => {
    const renderer = new ShallowRenderer();
    
    renderer.render(<Header/>);
    console.log(renderer.getRenderOutput());
    
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
