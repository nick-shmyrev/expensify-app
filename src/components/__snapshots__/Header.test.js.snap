// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <NavLink
      activeClassName="active"
      aria-current="page"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/create"
    >
      Create Expense
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/help"
    >
      Help Page
    </NavLink>
  </nav>
</header>
`;
